testing/src/main/java/io/grpc/internal/testing/AbstractTransportTest.java

```diff
@@ -1293,10 +1293,9 @@ public void flowControlPushBack() throws Exception {
     serverReceived +=
         verifyMessageCountAndClose(serverStreamListener.messageQueue, clientSent - serverReceived);
 
-    assertTrue(clientStreamListener.awaitOnReady(TIMEOUT_MS, TimeUnit.MILLISECONDS));
-    assertTrue(clientStreamListener.awaitOnReady(TIMEOUT_MS, TimeUnit.MILLISECONDS));
```

Comment:
I don't understand the previous code, was it duplicate lines and intended to be only one line?

Reply:
If you look back in a8db154 it was originally a mock verification with times(2). However, when it was changed it became a noop, since onReady was already previously called, but it _looked_ like it was checking twice.

Today I'm seeing onReady only called once, but the behavior isn't actually that precise since there aren't any other calls to notice if `isReady() == false`. So now I just require it to be called once, and if it happens twice that's "okay" as long it isn't racy, which could be detected via the isReady() call on the next line.